lib/cocoapods/downloader.rb

```diff
@@ -33,11 +33,16 @@ def self.download(
       cache_path: Config.instance.cache_root + 'Pods'
     )
       can_cache &&= !Config.instance.skip_download_cache
+
+      request = preprocess_request(request)
+
       if can_cache
         raise ArgumentError, 'Must provide a `cache_path` when caching.' unless cache_path
         cache = Cache.new(cache_path)
         result = cache.download_pod(request)
       else
+        raise ArgumentError, 'Must provide a `target` when caching is disabled.' unless target
```

Comment:
exactly. I stumbled upon it throwing a random error farther down the call chain so I copied the check for `cache_path`.

I've added a simple test btw. Considering the coverage over at `cocoapods-downloader` I think this is enough, but I can add more if you disagree.